analyses/compare-rnaseq-pipelines/utils/compare_RNASeq_pipelines.R

```diff
@@ -0,0 +1,58 @@
+# Invoke libraries
+suppressPackageStartupMessages(
+  {
+    library(tidyr)
+    library(dplyr)
+  }
+)
+
+# This function performs quality check on expected RNA-Seq counts input  matrices for which correlation coefficient can be calculated
+# Input: 2 matrices with RNA-seq counts processed by different pipelines
+# Output: R list object containing 2 matrices
+# Output_note: Each result matrix can be accessed as calc_cor_input[[mat1]]
+# Output_note: , calc_cor_input[[mat2]]
+qc_matrix_input <- function(matrix1, matrix2){
+  #rownames must refer to gene names
+  common_genes <- intersect(rownames(matrix1),rownames(matrix2)) %>%
+    sort()
+  mat1_all_genes <- matrix1[common_genes,]
+  mat2_all_genes <- matrix2[common_genes,]
+  
+  #colnames must refer to sample ids
+  common_samples <- intersect(colnames(mat1_all_genes), colnames(mat2_all_genes)) %>%
+    sort()
+  mat1_final <- mat1_all_genes[, common_samples]
+  mat2_final <- mat2_all_genes[, common_samples]
+  
+  #Validate all rows/genes and all columns/sample_ids match across matrices
+  mat1_col_extra <- setdiff(colnames(mat1_final),colnames(mat2_final))
+  # character(0)
+  mat2_col_extra <- setdiff(colnames(mat2_final),colnames(mat1_final))
+  # character(0)
```

Comment:
```suggestion
  #Validate all rows/genes and all columns/sample_ids match across matrices
  stopifnot(setequal((colnames(mat1_final), colnames(mat2_final)) == TRUE))
  stopifnot(setequal((rownames(mat1_final), rownames(mat2_final)) == TRUE))
```
This validation is not useful and has no effect on the script even if the row names and/or column names did not match.  Suggest to you remove it altogether because you already subsetted both matrices a few lines of code above to have identical column names and row names.

Reply:
@ewafula These lines were only meant for validating the columns match at the time of development. I am okay with removing them now.

Reply:
@ewafula While I updated the script, I have still kept the `stopifnot` logic so we can have error message for better handling.